internal/launcher/launcher_test.go

```diff
@@ -0,0 +1,284 @@
+package launcher
+
+import (
+	"errors"
+	"maps"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/k6deps"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"go.k6.io/k6/cmd/state"
+	"go.k6.io/k6/internal/build"
+	"go.k6.io/k6/internal/cmd/tests"
+)
+
+// launcherFixture mocks the functions used by the launcher and tracks their execution
+type luncherFixture struct {
+	provisionCalled bool
+	provisionDeps   k6deps.Dependencies
+	k6BinaryPath    string
+	k6Versions      string
+	provisionError  error
+	runK6Called     bool
+	runK6Error      error
+	runk6ReturnCode int
+	fallbackCalled  bool
+}
+
+func (f *luncherFixture) fallback(gs *state.GlobalState) {
+	f.fallbackCalled = true
+	gs.OSExit(0)
+}
+
+func (f *luncherFixture) provision(_ *state.GlobalState, deps k6deps.Dependencies) (string, string, error) {
+	f.provisionCalled = true
+	f.provisionDeps = deps
+	return f.k6BinaryPath, f.k6Versions, f.provisionError
+}
+
+// function to execute k6 binary
+func (f *luncherFixture) run(*state.GlobalState, string) (int, error) {
+	f.runK6Called = true
+	return f.runk6ReturnCode, f.runK6Error
+}
+
+const (
+	fakerTest = `
+import { Faker } from "k6/x/faker";
+
+const faker = new Faker(11);
+
+export default function () {
+  console.log(faker.person.firstName());
+}
+`
+
+	noDepsTest = `
+import http from 'k6/http';
+import { sleep, check } from 'k6';
+
+export default function() {
+  let res = http.get('https://quickpizza.grafana.com');
+  check(res, { "status is 200": (res) => res.status === 200 });
+  sleep(1);
+}
+`
+
+	requireUnsatisfiedK6Version = `
+"use k6 = v0.99"
+
+import { sleep, check } from 'k6';
+
+export default function() {
+  let res = http.get('https://quickpizza.grafana.com');
+  check(res, { "status is 200": (res) => res.status === 200 });
+  sleep(1);
+}
+`
+	// FIXME: when the build version is a prerelease (e.g v1.0.0-rc1), k6deps fails to parse this pragma
+	// and creates an invalid constrain that is ignored by the test.
+	// see https://github.com/grafana/k6deps/issues/91
```

Comment:
Would be possible to fix this before moving forward with this PR? 🤔 Although, I guess we could go without it, tho..

Reply:
That requires changes in the implementation of `k6deps` (https://github.com/grafana/k6deps/issues/91) Not sure if this is severe enough to block this PR.

Reply:
https://github.com/grafana/k6/issues/4694

Reply:
@joanlopez does the epic address your request?